Assets/MixedRealityToolkit/_Core/Utilities/UX/HoverLight.cs

```diff
@@ -92,21 +100,87 @@ private void OnDrawGizmosSelected()
             Gizmos.DrawIcon(transform.position + Vector3.back * Radius, string.Empty, false);
         }
 
-        private void Initialize()
+        private static void AddHoverLight(HoverLight light)
+        {
+            if (activeHoverLights.Count >= hoverLightCount)
+            {
+                Debug.LogWarningFormat("Max hover light count ({0}) exceeded.", hoverLightCount);
+            }
+
+            activeHoverLights.Add(light);
+        }
+
+        private static void RemoveHoverLight(HoverLight light)
+        {
+            activeHoverLights.Remove(light);
+        }
+
+        private static void Initialize()
+        {
+            _HoverLightDataID = Shader.PropertyToID("_HoverLightData");
+        }
+
+        private static void UpdateHoverLights(bool forceUpdate = false)
         {
-            hoverPositionId = Shader.PropertyToID("_HoverPosition");
-            hoverRadiusId = Shader.PropertyToID("_HoverRadius");
-            hoverColorId = Shader.PropertyToID("_HoverColor");
+            if (lastHoverLightUpdate == -1)
+            {
+                Initialize();
+            }
+
+            if (!forceUpdate && (Time.frameCount == lastHoverLightUpdate))
+            {
+                return;
+            }
+
+            if (activeHoverLights.Count > 1)
+            {
+                Shader.EnableKeyword("_MULTI_HOVER_LIGHT");
+            }
+            else
+            {
+                Shader.DisableKeyword("_MULTI_HOVER_LIGHT");
+            }
+
+            for (int i = 0; i < hoverLightCount; ++i)
+            {
+                HoverLight light = (i >= activeHoverLights.Count) ? null : activeHoverLights[i];
+                int dataIndex = i * hoverLightDataSize;
+
+                if (light)
+                {
+                    hoverLightData[dataIndex] = new Vector4(light.transform.position.x,
+                                                            light.transform.position.y,
+                                                            light.transform.position.z,
+                                                            light.Radius);
+                    hoverLightData[dataIndex + 1] = new Vector4(light.Color.r,
+                                                                light.Color.g,
+                                                                light.Color.b,
+                                                                1.0f);
+                }
+                else
+                {
+                    hoverLightData[dataIndex] = Vector4.zero;
+                    hoverLightData[dataIndex + 1] = Vector4.zero;
+                }
+            }
+
+            Shader.SetGlobalVectorArray(_HoverLightDataID, hoverLightData);
+
+            lastHoverLightUpdate = Time.frameCount;
         }
+    }
+
+#if UNITY_EDITOR
```

Comment:
Editors cannot exist inside a runtime script. They must be broken out in to their own files in the inspectors folder.
"/Assets/MixedRealityToolkit/_Core/Inspectors "

Reply:
I can certainly do this but, I probably won't have time until next week.

Reply:
It's ok @Cameron-Micka , it's for the vNext PR only.  @keveleigh , I'll send you a PR with the updates. (benefits of being ahead of time :D)

Reply:
As PR cancelled, ensure to move the Editor code to it's own class